pkg/addon/addon.go

```diff
@@ -1393,6 +1394,26 @@ func (h *Installer) continueOrRestartWorkflow() error {
 	return nil
 }
 
+func (h *Installer) getAddonVersionMeetSystemRequirement(addonName string) string {
+	if h.r != nil && IsVersionRegistry(*h.r) {
+		versionedRegistry := BuildVersionedRegistry(h.r.Name, h.r.Helm.URL, &common.HTTPOption{
+			Username: h.r.Helm.Username,
+			Password: h.r.Helm.Password,
+		})
+		versions, err := versionedRegistry.GetAddonAvailableVersion(addonName)
+		if err != nil {
+			return ""
+		}
+		for _, version := range versions {
+			req := LoadSystemRequirements(version.Annotations["system"])
+			if checkAddonVersionMeetRequired(h.ctx, req, h.cli, h.dc) == nil {
+				return version.Version
```

Comment:
If multiple addon versions match requirement. Should this function returns all of them, or just return one.
If it only returns one version, I think it should return the latest one.

Reply:
Yes, getAddonVersionMeetSystemRequirement return the addon's latest version which meet the system requirements